features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java

```diff
@@ -540,6 +600,120 @@ private BmpGlobalIpRib buildGlobalIpRib(PrefixByAS prefixByAS) {
+            bmpAsnInfo.setOrgName(asnInfo.getOrgName());
+            bmpAsnInfo.setAddress(asnInfo.getAddress());
+            bmpAsnInfo.setCity(asnInfo.getCity());
+            bmpAsnInfo.setStateProv(asnInfo.getStateProv());
+            bmpAsnInfo.setPostalCode(asnInfo.getPostalCode());
+            bmpAsnInfo.setCountry(asnInfo.getCountry());
+            bmpAsnInfo.setSource(asnInfo.getSource());
+            bmpAsnInfo.setRawOutput(asnInfo.getRawOutput());
+            bmpAsnInfo.setRemarks(asnInfo.getRemarks());
+            bmpAsnInfo.setLastUpdated(Date.from(Instant.now()));
+            return bmpAsnInfo;
+        }
+        return null;
+    }
+
+    private BmpRouteInfo fetchAndBuildRouteInfo(String prefix) {
+        Optional<RouteInfo> routeInfoOptional = BmpWhoIsClient.getRouteInfo(prefix);
+        if (routeInfoOptional.isPresent() && routeInfoOptional.get().getPrefix() != null) {
+            RouteInfo routeInfo = routeInfoOptional.get();
+            Integer prefixLen = routeInfo.getPrefixLen();
+            Long originAs = routeInfo.getOriginAs();
+            BmpRouteInfo bmpRouteInfo = bmpRouteInfoDao.findByPrefix(routeInfo.getPrefix(), prefixLen, originAs);
+            if (bmpRouteInfo == null) {
+                bmpRouteInfo = new BmpRouteInfo();
+                bmpRouteInfo.setPrefix(routeInfo.getPrefix());
+                bmpRouteInfo.setPrefixLen(routeInfo.getPrefixLen());
+                bmpRouteInfo.setDescr(routeInfo.getDescription());
+                bmpRouteInfo.setOriginAs(routeInfo.getOriginAs());
+                bmpRouteInfo.setSource(routeInfo.getSource());
+            }
+            bmpRouteInfo.setLastUpdated(Date.from(Instant.now()));
+            return bmpRouteInfo;
+        }
+        return null;
+    }
+
+    private List<BmpAsnPathAnalysis> buildBmpAsnPath(String asnPath) {
+
+        List<BmpAsnPathAnalysis> bmpAsnPathAnalyses = new ArrayList<>();
+        String[] asnArray = asnPath.split(" ");
+
+        Long leftAsn = 0L;
+        Long rightAsn = 0L;
+        Long asn = 0L;
+        for (int i = 0; i < asnArray.length; i++) {
+            if (asnArray[i].length() <= 0)
+                break;
+
+            try {
+                asn = Long.valueOf(asnArray[i]);
+            } catch (NumberFormatException e) {
+                e.printStackTrace();
+                break;
+            }
+
+            if (asn > 0) {
+                if (i + 1 < asnArray.length) {
+
+                    if (asnArray[i + 1].length() <= 0)
+                        break;
+
+                    try {
+                        rightAsn = Long.valueOf(asnArray[i + 1]);
+
+                    } catch (NumberFormatException e) {
+                        e.printStackTrace();
+                        break;
+                    }
+
+                    if (rightAsn.equals(asn)) {
+                        continue;
+                    }
+
+                    Boolean isPeeringAsn = (i == 0 || i == 1) ? TRUE : FALSE;
```

Comment:
This code path is never taken if `i == 0` because `asn` will always be `0` on the first iteration.